From 1b2e221a9854d0ae377b04478fe8000c138c3511 Mon Sep 17 00:00:00 2001
From: Marco Martin <notmart@gmail.com>
Date: Thu, 5 Jan 2012 16:59:05 +0100
Subject: [PATCH 1/2] contextview taskbar not child of containment

the containment being parent can control its appearance, such as clipping
BUG:290123
---
 src/context/ToolbarView.cpp           |    6 +++-
 src/context/toolbar/AppletToolbar.cpp |   34 +++++++++++++++++++++++++++++++++
 src/context/toolbar/AppletToolbar.h   |    3 ++
 3 files changed, 41 insertions(+), 2 deletions(-)

diff --git a/src/context/ToolbarView.cpp b/src/context/ToolbarView.cpp
index f3b4916..840b99b 100644
--- a/src/context/ToolbarView.cpp
+++ b/src/context/ToolbarView.cpp
@@ -74,7 +74,9 @@ Context::ToolbarView::ToolbarView( Plasma::Containment* containment, QGraphicsSc
     setVerticalScrollBarPolicy( Qt::ScrollBarAlwaysOff );
 
     // now we create the toolbar
-    m_toolbar = new AppletToolbar( containment );
+    m_toolbar = new AppletToolbar(0);
+    scene->addItem(m_toolbar.data());
+    m_toolbar.data()->setContainment( qobject_cast<Context::Containment *>(containment) );
     m_toolbar.data()->setZValue( m_toolbar.data()->zValue() + 1000 );
     m_toolbar.data()->setPos( TOOLBAR_X_OFFSET, 0 );
 
@@ -96,7 +98,7 @@ Context::ToolbarView::ToolbarView( Plasma::Containment* containment, QGraphicsSc
 
 Context::ToolbarView::~ToolbarView()
 {
-
+    delete m_toolbar.data();
 }
 
 void
diff --git a/src/context/toolbar/AppletToolbar.cpp b/src/context/toolbar/AppletToolbar.cpp
index 0e0deb5..c5d749b 100644
--- a/src/context/toolbar/AppletToolbar.cpp
+++ b/src/context/toolbar/AppletToolbar.cpp
@@ -67,6 +67,40 @@ Context::AppletToolbar::~AppletToolbar()
 }
 
 void
+<<<<<<< Updated upstream
+=======
+Context::AppletToolbar::setContainment( Containment * containment )
+{
+    m_cont = containment;
+}
+
+Context::Containment *
+Context::AppletToolbar::containment() const
+{
+    return m_cont;
+}
+
+void
+Context::AppletToolbar::paint( QPainter * painter, const QStyleOptionGraphicsItem * option, QWidget * widget )
+{
+    Q_UNUSED( option )
+    Q_UNUSED( widget )
+
+    painter->setRenderHint( QPainter::Antialiasing );
+    painter->save();
+
+    QColor col = PaletteHandler::alternateBackgroundColor();
+    qreal radius = 3;
+
+    QPainterPath outline;
+    outline.addRoundedRect( boundingRect(), radius, radius );
+    painter->fillPath( outline, QBrush( col ) );
+
+    painter->restore();
+}
+
+void
+>>>>>>> Stashed changes
 Context::AppletToolbar::resizeEvent( QGraphicsSceneResizeEvent * event )
 {
     debug() << "setting layout to" << QRectF( QPointF( 0, 0 ), event->newSize() );
diff --git a/src/context/toolbar/AppletToolbar.h b/src/context/toolbar/AppletToolbar.h
index ee8a208..058baae 100644
--- a/src/context/toolbar/AppletToolbar.h
+++ b/src/context/toolbar/AppletToolbar.h
@@ -55,6 +55,9 @@ class AppletToolbar : public QGraphicsWidget
         
         void appletRemoved( Plasma::Applet* applet );
 
+        void setContainment( Containment * containment );
+        Containment* containment() const;
+
     signals:
         void showApplet( Plasma::Applet* );
         void appletAddedToToolbar( Plasma::Applet* applet, int loc );
-- 
1.7.6

